aws_signing_helper/update.go

```diff
@@ -0,0 +1,205 @@
+	destFile, err := GetWriteOnlyCredentialsFile()
+	if err != nil {
+		log.Println("unable to get write-only AWS credentials file")
+		syscall.Exit(1)
+	}
+	defer destFile.Close()
+
+	// Create buffered writer with a buffer of maximum size
+	destFileWriter := bufio.NewWriterSize(destFile, math.MaxInt32)
+
+	var profileExist = false
+	var profileSection = "[" + profileName + "]"
+	// A variable that checks whether or not required fields are written to the destination file
+	newCredVisit := map[string]bool{"aws_access_key_id": false, "aws_secret_access_key": false, "aws_session_token": false}
+	accessKey := "aws_access_key_id = " + cred.AccessKeyId + "\n"
+	secretKey := "aws_secret_access_key = " + cred.SecretAccessKey + "\n"
+	sessionToken := "aws_session_token = " + cred.SessionToken + "\n"
+
+	for i := 0; i < len(writeLines); i++ {
+		if !profileExist && writeLines[i] == profileSection {
+			j := i + 1
+			destFileWriter.WriteString(profileSection + "\n")
+			for ; j < len(writeLines); j++ {
+				// If the last line of the credentials file is reached
+				// OR the next profile section is reached
+				if j == len(writeLines)-1 || strings.HasPrefix(writeLines[j], "[") {
+					if !newCredVisit["aws_access_key_id"] {
+						_, err = destFileWriter.WriteString(accessKey)
+					}
+					if !newCredVisit["aws_secret_access_key"] {
+						_, err = destFileWriter.WriteString(secretKey)
+					}
+					if !newCredVisit["aws_session_token"] {
+						_, err = destFileWriter.WriteString(sessionToken)
+
+					}
+					if j == len(writeLines)-1 {
+						i = j
+					} else {
+						i = j - 1
+					}
+					profileExist = true
+					break
+				} else if strings.HasPrefix(writeLines[j], "aws_access_key_id") {
+					// replace "aws_access_key_id"
+					_, err = destFileWriter.WriteString(accessKey)
+					newCredVisit["aws_access_key_id"] = true
+				} else if strings.HasPrefix(writeLines[j], "aws_secret_access_key") {
+					// replace "aws_secret_access_key"
+					_, err = destFileWriter.WriteString(secretKey)
+					newCredVisit["aws_secret_access_key"] = true
+				} else if strings.HasPrefix(writeLines[j], "aws_session_token") {
+					// replace "aws_session_token"
+					_, err = destFileWriter.WriteString(sessionToken)
+					newCredVisit["aws_session_token"] = true
+				} else {
+					// write other keys
+					_, err = destFileWriter.WriteString(writeLines[j] + "\n")
+				}
+			}
+		} else {
+			_, err = destFileWriter.WriteString(writeLines[i] + "\n")
+		}
+	}
+
+	// If the chosen profile does not exist
+	if !profileExist {
+		writeCredential := profileSection + "\n" + accessKey + secretKey + sessionToken
+		_, err = destFileWriter.WriteString(writeCredential + "\n")
+	}
+
+	if err != nil {
+		return err
+	}
+
+	// Flush the contents of the buffer
+	destFileWriter.Flush()
+	return nil
+}
+
+// Function to check whether credentials are still valid
+func ValidCred(cred TemporaryCredential) bool {
```

Comment:
Minor - `Valid` implies that the check is deeper than just checking expiry. Maybe `ExpiringCred` or `CredIsExpiring` convey the intent better?

Reply:
Oops, this function is unused. I have removed it in the next revision.